core/shared/src/main/scala/zio/Scope.scala

```diff
@@ -185,9 +167,39 @@ object Scope {
   def parallel(implicit trace: Trace): UIO[Scope.Closeable] =
     makeWith(ExecutionStrategy.Parallel)
 
+  object unsafe {
+    def make(implicit unsafe: Unsafe): Scope.Closeable =
+      new CloseableScope(ExecutionStrategy.Sequential)
+
+    def makeWith(executionStrategy: ExecutionStrategy)(implicit unsafe: Unsafe): Scope.Closeable =
+      new CloseableScope(executionStrategy)
+
+    private final class CloseableScope(override val executionStrategy: ExecutionStrategy) extends Scope.Closeable {
+      private[this] val releaseMap = new ReleaseMap
```

Comment:
This can extend `ReleaseMap` if you mark it non-final 😅

Reply:
The issue with this is that we'll have to do is as `class CloseableScope(...) extends ReleaseMap with Scope.Closeable`. There is a bug in the JDK that was resolved in JDK23 where type-checking on a secondary super doesn't scale. Since we're using `isInstanceOf[Scope]` within `ZEnvironment`, I'm a bit reluctant to make this change. WDYT?

Reply:
Interesting - I didn't that. We can keep it as is for now.